Remotion/Web/Development.WebTesting.IntegrationTests.Infrastructure/GenericTestParameterConverter.cs

```diff
@@ -1,69 +1,108 @@
+// and/or modify it under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation; either version 2.1 of the
 // License, or (at your option) any later version.
-// 
-// re-motion is distributed in the hope that it will be useful, 
-// but WITHOUT ANY WARRANTY; without even the implied warranty of 
-// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the 
+//
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 // GNU Lesser General Public License for more details.
-// 
+//
 // You should have received a copy of the GNU Lesser General Public License
 // along with re-motion; if not, see http://www.gnu.org/licenses.
-// 
+//
 using System;
-using System.Collections;
 using System.Collections.Generic;
 using System.Linq;
-using System.Web.Script.Serialization;
+using System.Text.Json;
+using System.Text.Json.Serialization;
 
 // ReSharper disable once CheckNamespace
 
 namespace Remotion.Web.Development.WebTesting.IntegrationTests.Infrastructure
 {
-  public class GenericTestParameterConverter : JavaScriptConverter
+  public class GenericTestParameterConverter : JsonConverter<GenericTestPageParameterDto>
   {
-    public static readonly GenericTestParameterConverter Instance = new GenericTestParameterConverter();
-
-    /// <inheritdoc />
-    public override IEnumerable<Type> SupportedTypes { get; } = new[] { typeof (GenericTestPageParameterDto) };
+    public static readonly GenericTestParameterConverter Instance = new();
 
     private GenericTestParameterConverter ()
     {
     }
 
-    /// <inheritdoc />
-    public override object Deserialize (IDictionary<string, object> dictionary, Type type, JavaScriptSerializer serializer)
+    public override GenericTestPageParameterDto Read (ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+    {
+      if (reader.TokenType != JsonTokenType.StartObject)
+        throw CreateExceptionForExpectedTokenType (JsonTokenType.StartObject, reader);
+
+      reader.Read();
+
+      if (reader.TokenType != JsonTokenType.PropertyName)
+        throw CreateExceptionForExpectedTokenType (JsonTokenType.PropertyName, reader);
+
+      if (reader.GetString() != "status")
+        throw CreateExceptionForExpectedTokenStringValue ("status", reader);
+
+      reader.Read();
+      var status = (GenericTestPageStatus) reader.GetInt32();
+
+      reader.Read();
+
+      if (reader.TokenType != JsonTokenType.PropertyName)
+        throw CreateExceptionForExpectedTokenType (JsonTokenType.PropertyName, reader);
+
+      if (reader.GetString() != "parameters")
+        throw CreateExceptionForExpectedTokenStringValue ("parameters", reader);
+
+      reader.Read();
+
+      var dictionary = JsonSerializer.Deserialize<Dictionary<string, IList<string>>> (ref reader);
```

Comment:
Is it maybe useful to pass the `JsonSerializerOptions` given by the `options` parameter into this method or plain unnecessary?

Reply:
I don't think the `GenericTestParameterDto` will ever be nested within itself, but it still is best practice because of other possible options set. I changed it.